### DISCUSSION TITLE
Problem with two cameras (raspberrypi camera and C270 [USB])

### DISCUSSION THREAD

**Discussion by nilbest** [mainsail-crew/mainsail#1878] [Help]
What happened?
I set up both cameras yesterday, it was all fine.
Pi Cam as /dev/video0
C270 as /dev/v4l/by-id/usb-046d_0825_B862FCA0-video-index0
Today I started klipper and the Pi again and now some how both cameras where still detected via the logs but got switched internally around.
Log:
Pi Cam -> /dev/video2
C270 -> /dev/video0 but still as /dev/v4l/by-id/usb-046d_0825_B862FCA0-video-index0
And the Pi Camera got no signal.
I used both in ustreamer mode and everything was fine with good fps (around 15).

I changed them around and the C270 was fine again.
But to get the Pi Cam back I have to use camera-streamer, but then both video feeds have low fps at under 5fps.

What did you expect to happen?
I expected both cameras would have the same video{Number} then yesterday because I didn't change anything.
They would be directly callable via /dev/video0 and as /dev/v4l/by-id/usb-046d_0825_B862FCA0-video-index0 for the C270.
I expected the same fps performance then yesterday and the same modes for both of them.

How to reproduce:
I'm not sure.
Happens to me sometimes when I power everything off for one day and would get back on it.
(And Yes I did a shutdown from mainsail UI and waited around 5min before pulling the power plug)

Additional information:
Raspberry Pi 4 B 2Gb
Logitec C270
Raspberry Pi Camera Module 5MP OV5647

crownsnest v4.1.9-gd75a3aeb
klipper v0.12.0-179-g434770ea
mainsail v2.11.2
mainsail-config v1.2.1-0-ge57810d5
moonraker v0.8.0-354-gba942855
sonar v0.1.3-1-geacc498a
System OS-Packages

**Comment by mryel00** [comment]
Might be a bit late, as I don't monitor the discussions actively. But here some explanation on this:
You are using the legacy stack right now, if this is an official picam, you should switch to libcamera following [this](https://crowsnest.mainsail.xyz/faq/how-to-setup-a-raspicam#raspicam-not-detected)
Otherwise there are only some workarounds in place that don't work reliable. Iirc you have ti unplug the usb cam. Then reboot. Then the device should be back on `/dev/video0`. Then reboot again and it should stay there even after plugging in the other cam again and then rebooting.

I'm not entirely sure if that is still in place or if it worked consistently.
